CIP-0161/README.md

```diff
@@ -0,0 +1,1871 @@
+
+To maintain this level of security over time:  
+- Governance may choose to **increase the 12-hour budget** as the cost of computation decreases.  
+- Execution could migrate to **ASIC-based architectures**, preserving the same budget while maintaining security guarantees, and later increasing the budget further.  
+
+Beyond parameter updates, adoption of this proposal would necessarily require a **hard fork**, since it modifies the consensus protocol in two fundamental ways:  
+1. The randomness for slot distribution is extended from **1 epoch to 2 epochs**. At the start of epoch *e*, the snapshot of the stake distribution will be taken at the end of *epoch e−2*, rather than at the end of *epoch e−1* as in Praos today.  
+2. The **general method of generating slot leader distributions** is changed, making leader election more resilient to adversarial bias.
+
+### Consensus Performance
+
+This proposal is not only about security, but also about **consensus performance**.  
+
+In Praos, because grinding allows adversaries to choose among multiple possible slot leader distributions, the probability of “bad events” (such as rollbacks or settlement failures) is statistically amplified compared to the honest model.  
+
+- If a bad event occurs with probability $`\varepsilon`$ under unbiased randomness,  
+- An adversary able to try $`R`$ independent randomness candidates can increase the likelihood of that event up to $`R \cdot \varepsilon`$ (by the union bound).  
+
+This translates into slower settlement and weaker guarantees for the network as a whole. By substantially reducing $`R`$ compared to Praos, we limit the impact of grinding attacks and therefore improve settlement. In fact, the recommended configuration reduces settlement time by approximately **20–30%** while maintaining equivalent security.
+
+### Relationship to Peras
+
+[Ouroboros Peras](https://peras.cardano-scaling.org/) is a recent extension of Praos designed to **accelerate settlement**.  
+- Instead of waiting for the traditional 2160-block window (around 5 days) to guarantee finality, Peras introduces **stake-weighted voting and certified blocks**.  
+- Randomly chosen committees of stake pool operators can “vote” on blocks, and when enough votes are collected, the block receives a certificate.  
+- Certified blocks are treated as more important in the chain, which enables **settlement in just 1–2 minutes**.
+
+Peras is fully compatible with Praos:  
+- When enough committee members participate, it achieves **rapid settlement**.  
+- When they do not (e.g., if too many operators are offline), Peras **gracefully falls back to Praos**.  
+
+In these fallback situations, the network still relies on Praos’ guarantees—precisely where Phalanx becomes relevant as a **complementary defense layer**. Phalanx ensures that even when Peras cannot certify blocks, the protocol still benefits from:  
```

Comment:
im curious
do we know the likelihood of these fallback situations?

Reply:
Peras would only fall back to Praos if there were massive network disruption or an attack by a 25% adversary.

Reply:
I will add it